src/pveclib/vec_int32_ppc.h

```diff
@@ -880,6 +978,17 @@ vec_muleuw (vui32_t a, vui32_t b)
  * Multiple the odd words of two vector unsigned int values and return
  * the unsigned long product of the odd words.
  *
+ * For POWER8 and later we can use the vmulouw instruction.
+ * But for POWER7 and earlier we have to construct word multiplies
+ * from two halfword multiplies (vmuleuh and vmulouh). Then sum the
+ * partial products for the final doubleword results. This is
+ * complicated the fact that vector add doubleword is not available
+ * for POWER7. So we need to construct the doubleword add from
+ * Vector Add Unsigned Word Modulo (vadduwm) and
+ * Vector Add and Write Carry-Out Unsigned Word (vaddcuw) with
+ * shift double quadword to reposition the low word carry and
+ * a final vadduwm to complete carry propagation the doubleword add.
```

Comment:
...complete _the_ carry propagation _for_ the doublework add. ?

Reply:
Ok